OvmfPkg/QemuFlashFvbServicesRuntimeDxe/QemuFlash.c

```diff
@@ -259,6 +259,14 @@ QemuFlashInitialize (
   VOID
   )
 {
+  //
+  // The SNP model does not provide for QEMU flash device support, so exit
+  // early before attempting to initialize any QEMU flash device support.
+  //
+  if (MemEncryptSevSnpIsEnabled ()) {
+    return EFI_WRITE_PROTECTED;

```

Comment:
Minor nit: Shouldn't this be `EFI_UNSUPPORTED` (as in "flash not supported")?
I think qemu simply loads the firmware into RAM with SEV-SNP (TDX too), so it's not actually write protected ...